packages/neuron-wallet/src/services/tx/transaction-service.ts

```diff
@@ -76,32 +78,27 @@ export class TransactionsService {
 
     if (type === SearchType.Address) {
       const lockHashToSearch = AddressParser.parse(searchValue).computeHash()
-      lockHashes = [lockHashToSearch]
-      allTxHashes = await repository
-        .createQueryBuilder('tx')
-        .select('tx.hash', 'txHash')
-        .where(
-          `tx.hash
-        IN
-          (
+      allTxHashes = await connection
+        .createQueryRunner()
+        .query(
+          `
             SELECT transactionHash from (
-              SELECT output.transactionHash FROM output WHERE output.lockHash in (:...lockHashes)
+              SELECT output.transactionHash FROM output WHERE output.lockHash = @0
               UNION
-              SELECT input.transactionHash FROM input WHERE input.lockHash in (:...lockHashes)
+              SELECT input.transactionHash FROM input WHERE input.lockHash = @0
+              UNION
+              SELECT tx_lock.transactionHash FROM tx_lock WHERE tx_lock.lockHash = @0
```

Comment:
How about separating this query into 2 types

```sql
SELECT output.transactionHash from output WHERE output.lockHash = @0
UNION
SELECT input.transactionHash from input WHERE input.lockHash = @0
```
and
```sql
SELECT tx_lock.transactionHash from tx_lock WHERE tx_lock.lockHash = @0
```

And add a condition of `lockHash in current wallet` before this query to boost it.

`tx_lock` could be huge to introduce performance impact while it should not be queried so often as `current wallet`

By doing this, the `intersect` could be avoided too.

Reply:
`lockHash` is an index of `tx_lock` and `SELECT tx_lock.transactionHash from tx_lock WHERE tx_lock.lockHash = @` is filtered by whether `lockHash` is equal, so it should execute fast.